### PR TITLE
User Auth plugin trigger

### PR DESCRIPTION
**Pull request by josegonzalez**
I came up with a proof of concept for limiting commands to a given user. I'm not convinced this is the absolute best way, but I do think that it's the most flexible approach given how dokku is structured.

### CODE REVIEW

**Comment by josegonzalez** [comment]
@michaelshobbs thoughts on this?

File: dokku [hunk 1]
```diff
@@ -50,8 +50,9 @@ fi
 ! has_tty && DOKKU_QUIET_OUTPUT=1
 
 if [[ $(id -un) != "dokku" && $1 != plugin:*install* && $1 != "plugin:update" ]]; then
+  export SSH_USER=$(id -un)
   sudo -u dokku -E -H $0 "$@"
-  exit
+  exit $?
```
Comment (michaelshobbs) [inline dokku]: is this change to `exit` pertinent to this change? (just curious)

**Comment by michaelshobbs** [comment]
seems fine for a first pass

File: dokku (hunk 1 above)
Comment (josegonzalez) [inline dokku]: Nope, just wanted to change it while i was there.

Comment (michaelshobbs) [inline dokku]: :+1:

**Pull request merged**